src/otx/algorithms/common/adapters/torch/dataloaders/samplers/otx_sampler.py

```diff
@@ -0,0 +1,98 @@
+"""OTX sampler."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import math
+
+from torch.utils.data import Dataset
+from torch.utils.data.sampler import Sampler
+
+from otx.algorithms.common.utils.logger import get_logger
+from otx.algorithms.common.utils.task_adapt import unwrap_dataset
+
+logger = get_logger()
+
+
+class OTXSampler(Sampler):  # pylint: disable=too-many-instance-attributes
+    """Sampler that easily adapts to the dataset statistics.
+
+    In the exterme small dataset, the iteration per epoch could be set to 1 and then it could make slow training
+    since DataLoader reinitialized at every epoch. So, in the small dataset case,
+    OTXSampler repeats the dataset to enlarge the iterations per epoch.
+
+    In the large dataset, the useful information is not totally linear relationship with the number of datasets.
+    It closes to the log scale relationship, rather.
+
+    So, this sampler samples or repeats the datasets acoording to the statistics of dataset.
+
+    Args:
```

Comment:
It seems that argument comment is a little different with real arguments. Could you check it?

Reply:
done https://github.com/openvinotoolkit/training_extensions/pull/2428/commits/a496a82861ea094fca937a3c06caff9aac85f3fd